### PR TITLE
test: add unit-test and hack-verify-gomod

### PR DESCRIPTION
**Pull request by Sakuralbj**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/azuredisk/azure_common_test.go [hunk 1]
```diff
@@ -180,6 +214,35 @@ func TestNormalizeStorageAccountType(t *testing.T) {
...
+			expectedErr: fmt.Errorf("azureDisk - WriteOnly is not supported cachingmode. Supported values are [None ReadOnly ReadWrite]"),
+		},
+		{
+			desc:        "Valid CachingMode",
+			req:         "ReadOnly",
+			expectedErr: nil,
+		},
+	}
+	for _, test := range tests {
+		_, err := normalizeCachingMode(test.req)
```
Comment (andyzhangx) [inline pkg/azuredisk/azure_common_test.go:240]: pls also check return value, not only error

File: pkg/azuredisk/azure_common_test.go [hunk 2]
```diff
@@ -135,6 +139,36 @@ func TestIoHandler(t *testing.T) {
...
+			expectedErr: fmt.Errorf("azureDisk - WriteOnly is not supported disk kind. Supported values are [Dedicated Managed Shared]"),
+		},
+		{
+			desc:        "Valid CachingMode",
+			req:         string(api.AzureDedicatedBlobDisk),
+			expectedErr: nil,
+		},
+	}
+	for _, test := range tests {
+		_, err := normalizeKind(test.req)
```
Comment (andyzhangx) [inline pkg/azuredisk/azure_common_test.go:165]: pls also check return value, not only error

File: pkg/azuredisk/azure_common_test.go (hunk 1 above)
Comment (Sakuralbj) [reply pkg/azuredisk/azure_common_test.go:240]: ok

**Pull request merged**